src/test/util_string_tests.cpp

```diff
@@ -11,18 +11,30 @@ using namespace util;
 
 BOOST_AUTO_TEST_SUITE(util_string_tests)
 
+template <unsigned NumArgs>
+std::string TfmFormatZeroes(const char* fmt)
+{
+    return std::apply([fmt](auto... args) {
+        return tfm::format(std::string{fmt}, args...);
+    }, std::tuple_cat(std::array<int, NumArgs>{}));
+}
+
 // Helper to allow compile-time sanity checks while providing the number of
 // args directly. Normally PassFmt<sizeof...(Args)> would be used.
 template <unsigned NumArgs>
-inline void PassFmt(util::ConstevalFormatString<NumArgs> fmt)
+inline void PassFmt(ConstevalFormatString<NumArgs> fmt)
 {
-    // This was already executed at compile-time, but is executed again at run-time to avoid -Wunused.
+    // Exercise happy paths at run-time for code coverage metrics.
     decltype(fmt)::Detail_CheckNumFormatSpecifiers(fmt.fmt);
+
+    // Prove parity with tinyformat, which will throw if the number of
+    // conversion specifiers is incorrect.
+    TfmFormatZeroes<NumArgs>(fmt.fmt);
```

Comment:
I was wondering how it will behave with `%c` which would add a `\0` which could end the const char* prematurely, but since we have a string it seems to work correctly.

Are there any checks we could do with the result here? Or make `TfmFormatZeroes` void if we don't need the actual result. Maybe something like this:
```suggestion
    auto result{TfmFormatZeroes<NumArgs>(fmt.fmt)};
    BOOST_CHECK_EQUAL(result.empty(), std::string_view{fmt.fmt}.empty());
```

Reply:
re: https://github.com/bitcoin/bitcoin/pull/30933#discussion_r1869004667

I think suggested check would be broken in the case of `strprintf("%.0s", 0)` where result should be empty (because it is printing the first 0 characters of a string) even though the format string empty. It has hard to think of a reliable check that could be done here that would also be meaningful.

Reply:
We can void the method in that case